### ISSUE TITLE
Calling incorrect filename for content-product-cat.php

### ISSUE THREAD

**Issue by Josh-Millard** [woothemes/woocommerce#31380]
Please provide us with the information requested in this bug report. Without these details, we won't be able to fully evaluate this issue. 
Bug reports lacking detail, or for any other reason than to report a bug, may be closed without action.

**Prerequisites (mark completed items with an [x]):**
- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
I have recently taken on a project that needs Woocommerce implemented. The theme is built on Sage which uses Laravels blade templating system. I have included the App/Woocommerce.php in the Sage required files and managed to get my own version of the archive-product.php loading as expected. 

My issue is when I change the Woocoommerce shop page display to 'Categories', When the filter Woocommerce loop starts, the filter which is;

`add_filter( 'woocommerce_product_loop_start', 'woocommerce_maybe_show_product_subcategories' );`

Calls the following function 

`woocommerce_output_product_categories`

During this loop;

The template it looks for it content-product_cat.php, but in the Woocommerce/templates the file is named content-product-cat.php

**Expected behavior**
Function should call the correct filename of the template part - content-product-cat.php instead of content-product_cat.php

**Actual behavior**
Described in the 'bug behaviour' above.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Described in the 'bug behaviour' above.

**Screenshots**
N/A

**Isolating the problem (mark completed items with an [x]):**
- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
N/A

**Comment by peterfabian** [comment]
Hi, thanks for reporting. We actually renamed the templates some time back to bring them up to date with WordPress naming standards. For this reason, we've added a workaround where we [replace underscore with a dash in file names](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-core-functions.php#L392) in the `wc_locate_template` function, so theoretically speaking, this shouldn't matter much.

I'm wondering if you see the template not being loaded and thus content missing in your case or do you see a notice in the log file or what is actually happening for the store you're developing?

**Comment by Josh-Millard** [comment]
Hi, Thanks for your reply.

The template loaded fine when I changed the file name to content-product_cat.php, which is still the default template name in the Woocommerce templates). I had no error or notice, Just a blank page. All content loaded when that underscore was added.

https://github.com/roots/sage-woocommerce/blob/master/src/woocommerce.php

This is the extension I added to the Laravel theme setup which handles all the WC templates, Looking back at this now I cannot see the underscores/dashes being used.

**Comment by barryhughes** [comment]
In addition to Peter's note, I find I am unable to reproduce the problem (at least with a default theme). Here's what I tried:

- Set *"Shop page display"* to *"Show categories"*
- Set *"Category display"* to *"Show subcategories*"
- Viewed the main shop page, which rendered as expected
- Using breakpoints I confirmed that, when the above happens, the `content-product-cat.php` template is loaded successfully by function `woocommerce_output_product_categories()`

Given that, I'm not sure there is much more for us to do.

> The theme is built on Sage which uses Laravels blade templating system. I have included the App/Woocommerce.php in the Sage required files and managed to get my own version of the archive-product.php loading as expected.

Is it possible something in this site was causing problems that we wouldn't be able to see using a default setup?  Or, are you are able to distill this down to a set of steps-to-replicate we can use that requires just WooCommerce and a default theme?

Thanks!
